test/model.test.js

```diff
@@ -0,0 +1,893 @@
+'use strict';
+
+var Promise = require('bluebird'),
+    TestFixture = require('./test-fixture'),
+    expect = require('chai').expect,
+    errors = require('../lib/errors'),
+    util = require('./util');
+
+var test = new TestFixture(),
+    thinky, type;
+
+before(function() {
+  return test.initializeDatabase()
+    .then(function() {
+      thinky = test.thinky;
+      type = thinky.type;
+    });
+});
+after(function() { return test.dropDatabase(); });
+
+describe('createModel', function() {
+  afterEach(function() { return test.dropTables(); });
+
+  it('should create a new model', function() {
+    var model = thinky.createModel(util.s8(), {
+      id: String, name: String
+    });
+
+    return model.ready()
+      .then(function() { expect(model).to.exist; });
```

Comment:
`model` is immediately defined. There's no need to wait for the promise returned by `ready`